lib/queue_classic_plus/tasks/work.rake

```diff
@@ -16,19 +16,17 @@ namespace :qc_plus do
     trap('INT') do
       $stderr.puts("Received INT. Shutting down.")
       if !@worker.running
-        $stderr.puts("Worker has stopped running. Exit.")
-        exit(1)
+        $stderr.puts("Worker has already stopped running.")
       end
       @worker.stop
```

Comment:
`stop` actually just sets an ivar, so it's definitely safe: https://github.com/QueueClassic/queue_classic/blob/master/lib/queue_classic/worker.rb#L57

I think that's where the confusion stems from: `stop` doesn't actually block and stop the worker, it just prevents it from picking up new jobs.